webapp/lib/controller.dart

```diff
@@ -90,6 +96,47 @@ void initUI() {
   );
 }
 
+void setupProjectTimer(String project) {
+  var timer = new Timer(Duration(hours: 1), () => view.contentView.stale = true);
+  projectTimers[project] = timer;
+}
+
+model.NeedsReplyData getLatestDataForSelectedProject(List<model.NeedsReplyData> updatedData, String selectedProjectName) {
+  var latestProjectData = [];
+  latestProjectData.addAll(updatedData.where((data) => data.project == selectedProjectName));
+  if (latestProjectData.isEmpty) {
+    return null;
+  } else {
+    return latestProjectData.last;
+  }
+}
+
+void checkNeedsReplyMetricsStale(List<model.NeedsReplyData> updatedData) {
+  updatedData.sort((d1, d2) => d1.datetime.compareTo(d2.datetime));
+  
+  String selectedProjectName = view.contentView.projectSelectorView.selectedProject;
+  var latestSelectedProjectData = getLatestDataForSelectedProject(updatedData, selectedProjectName);
+
+  if (latestSelectedProjectData != null) {
+    var now = new DateTime.now();
+    int lastUpdateTimeDiff =  now.difference(latestSelectedProjectData.datetime).inHours;
+    
+    if (lastUpdateTimeDiff >= 1) {
+        view.contentView.stale = true;
+      } else {
+        view.contentView.stale = false;
+      }
+  } else {
+    view.contentView.stale = true;
+  }
+
+  var currentProjectTimer =  projectTimers[selectedProjectName];
+  if (currentProjectTimer != null) {
+    currentProjectTimer.cancel();
+  }
+  setupProjectTimer(selectedProjectName);
```

Comment:
we should update not just the selected project timer, but all of the timers that have received new updated data

Reply:
I also think it would be less error-prone to cancel the timer in `setupProjectTimer` instead of here, outside it. That way, `setupProjectTimer` takes care of everything, and if we need to call it somewhere else, we don't need to duplicate it

Reply:
Oh....I had thought we said we're only updating the timer for the selected project. But it makes sense creating/updating timers for all projects so that when the project is changed in the dropdown there is actually a timer to check